src/python/pants/backend/terraform/goals/check.py

```diff
@@ -11,28 +10,19 @@
 from pants.engine.process import FallibleProcessResult
 from pants.engine.rules import collect_rules, rule
 from pants.engine.unions import UnionRule
+from pants.option.option_types import BoolOption
 from pants.option.subsystem import Subsystem
 
 
 class TerraformValidateSubsystem(Subsystem):
     options_scope = "terraform-validate"
     help = """Terraform validate options."""
 
-    @classmethod
-    def register_options(cls, register):
-        super().register_options(register)
-        register(
-            "--skip",
-            type=bool,
-            default=False,
-            help=(
-                f"Don't run `terraform validate` when running `{register.bootstrap.pants_bin_name} check`."
-            ),
-        )
-
-    @property
-    def skip(self) -> bool:
-        return cast(bool, self.options.skip)
+    skip = BoolOption(
+        "--skip",
+        default=False,
+        help="Don't run `terraform validate` when running `./pants check`.",
```

Comment:
This is up for debate right now.
https://pantsbuild.slack.com/archives/C0D7TNJHL/p1644025386214639

(same below)